src/renderer/actions/queries.js

```diff
@@ -169,7 +176,7 @@ export function saveQuery () {
         { name: 'All Files', extensions: ['*'] },
       ];
 
-      let filename = (currentQuery.filename || await fileHandler.showSaveDialog(filters));
+      let filename = getFileName(currentQuery, isSaveAs, filters);
```

Comment:
Need to add an `await` here as `getFileName` returns `Promise<string>`. Without it, get and error where `filename` is a promise object and passing that to `fileHandler.saveFile` breaks.

```suggestion
      let filename = await getFileName(currentQuery, isSaveAs, filters);
```

Reply:
Good catch!!